### PR TITLE
Docs: Describe how to create higher-level construct

### PR DESCRIPTION
**Pull request by Doug-AWS**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

I've added a blurb on creating a "higher-level" construct from a basic CFN resource. Feel free to suggest changes.

### CODE REVIEW

File: docs/src/cloudformation.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@
 AWS CloudFormation Library
 ##########################
 
-The :doc:`AWS Construct Library <aws-construct-lib>` includes constructs with rich APIs
```
Comment (eladb) [inline docs/src/cloudformation.rst]: Why did you remove the link?

Comment (Doug-AWS) [reply docs/src/cloudformation.rst]: The link was originally in a different topic/RST file. We renamed aws-construct-lib.rst as cloudformation.rst and moved the text (including the link) into cloudformation.rst. Fortunately, if Sphinx cannot find a link, it just leaves the text. Not sure if we get any Sphinx error/warning during the build.

Comment (eladb) [reply docs/src/cloudformation.rst]: Nope, aws-construct-lib is a totally different topic:

- https://awslabs.github.io/aws-cdk/cloudformation.html
- https://awslabs.github.io/aws-cdk/aws-construct-lib.html

Comment (Doug-AWS) [reply docs/src/cloudformation.rst]: I'll add it back.
